modules/apps/portal-template/portal-template-react-renderer-impl/src/main/java/com/liferay/portal/template/react/renderer/internal/ReactRendererUtil.java

```diff
@@ -133,7 +144,32 @@ private static void _renderJavaScript(
 				_prepareData(componentDescriptor, data, httpServletRequest)));
 		javascriptSB.append(", '");
 		javascriptSB.append(placeholderId);
-		javascriptSB.append("');");
+		javascriptSB.append("'");
+
+		if (Validator.isNotNull(additionalProps)) {
+			String additionalPropsVarName = "additionalProps" + placeholderId + ".default";
+
+			javascriptSB.append(", typeof ");
```

Comment:
I've kept here the dual API just to keep our options open... but based on what I'm seeing chances of the static object being used are slim. I'd say 20% might use it since they don't require any param passing.

Reply:
If we go with `propsTransformer` or something of the like, then I don't think we can make sense of a static object and it seems more obvious a method is needed?

Reply:
👍 
If we are calling it a "transformer" I have no problem with it only being a function. Doesn't make sense for a transformer to be static.